src/buza/views.py

```diff
@@ -118,6 +118,30 @@ class SubjectDetail(generic.DetailView):
 class SubjectList(generic.ListView):
     model = models.Subject
 
+    # Order subjects that the user is following first, then by title.
+    ordering = ['-user_following', 'title']
+
+    def get_queryset(self) -> QuerySet:
+        """
+        Annotate each `Subject` with a `user_following` field,
+        which is true if the authenticated user is following it.
+        """
+        queryset: QuerySet = super().get_queryset()
+        user: models.RequestUser = self.request.user
+
+        if user.is_authenticated:
+            # Subquery existence check for relation to user:
+            return queryset.annotate(
```

Comment:
so to annotate is to add an extra field, metadata to a model. In this case we are adding a metafield to the each if the subjects called `user_following` which is true or false depending on whether the user is in the list of users that the follow the subject. The list of subjects is then ordered by the meta-field and title.

Reply:
this is actually mind blowing

Reply:
Yup! The annotation translates to an extra `SELECT` clause and subselect in the underlying SQL query, which the RDBMS can use for ordering and so on, and which Django also ends up mapping back to the returned Python model instances (though we are only using the RDBMS-level ordering, for now).